src/tests/unit/sync/syncimages_ring.f90

```diff
@@ -0,0 +1,25 @@
+! SYNC IMAGES([this_image - 1, this_image + 1]) with the STAT=STAT_STOPPED_IMAGE
+! specifier and wrap around of image numbers.  The test is intended to check
+! that syncing in a ring with a stopped image still terminates all images.
```

Comment:
> That there is an abort to be expected can be reflected in the name: sync_image_ring_abort_on_stopped_image

Yes, I meant rename the test source file.

>Idea: A test sync_image_ring should be modelled @rouson where the abort is not expected and it is just tested whether a ring is syncing.

This is a good idea. I'm trying to think of a good way to test this. My current thinking is that we could space images equally along a domain, say [0, 2pi), have each compute a periodic function (e.g. sine) at their location, and also perform a put to the left and right of what their neighbors value should be. (puts are asynchronous so the ring sync will actually be tested...) Then we do the ring sync and verify that the value put from the lhs neighbor and rhs neighbor match the local value (to within some suitable epsilon).

> It is always a good idea when the tests are not written by the dev who implemented a feature, because those tests tend to comply, because the dev knows what to do.

This is very true. Also tools like quick check that can randomly generate tests to check the properties of the API rather than hard-coded values, which is a similar idea.